### ISSUE TITLE
Sometimes transport_manager_test may fail with core crash

### ISSUE THREAD

**Issue by dboltovskyi** [smartdevicelink/sdl_core#3807]
### Bug Report
Sometimes transport_manager_test may fail with core crash

##### Reproduction Steps
1. Build SDL core with `-DBUILD_TESTS=ON` cmake option
2. Run `transport_manager_test` with command:
```
./transport_manager_test --gtest_filter=WebsocketNotSecureSessionConnectionTest.SUCCESS_SendData --gtest_repeat=100
```

##### Expected Behavior
Test passed

##### Observed Behavior
Test failed and core crash observed

##### OS & Version Information
* OS/Version: Ubuntu 18.04
* SDL Core Version: https://github.com/smartdevicelink/sdl_core/tree/develop (31324d5)
* Testing Against: N/A

##### Test Case, Sample Code, and / or Example App
 - [backtrace.txt](https://github.com/smartdevicelink/sdl_core/files/7435391/backtrace.txt)

**Comment by JackLivio** [comment]
@dboltovskyi Does Luxoft plan on submitting a pr for this issue?

**Comment by dboltovskyi** [comment]
@JackLivio Yes, Luxoft teams is going to submit a fix for this issue.

**Comment by jacobkeeler** [comment]
Closed via #3809
